code/segmentedtickerplant/datastripe.q

```diff
@@ -1,60 +1,64 @@
 
 configload:{
-     scpath:first .proc.getconfigfile[string .ds.segmentconfig];
-     fmpath:first .proc.getconfigfile[string .ds.filtermap];
-     {if[()~key hsym x;.lg.e[`init;"The following file can not be found: ",string x]]} each (scpath;fmpath);
-     @[{.stpps.segmentconfig:("SIS";enlist",")0: hsym x};scpath;{.lg.e[`init;"Failure in loading ",string y]}[;scpath]];
-     @[{.stpps.segmentfiltermap:(!/)(("S*";enlist",")0: hsym x)`wcRef`filter};fmpath;{.lg.e[`init;"Failure in loading ",string y]}[;fmpath]];
+     scpath:first .proc.getconfigfile[string .ds.stripeconfig];
+     {if[()~key hsym x;.lg.e[`init;"The following file can not be found: ",string scpath]]};
+     .stpps.stripeconfig:.j.k raze read0 scpath;
+     defaults:{first (flip .stpps.stripeconfig[x])[`subscriptiondefault]}each key .stpps.stripeconfig;
+     errors:1+ where {[x] not ("ignore"~x) or ("all"~x)}each defaults;
+     {if[0<count errors;.lg.e[`sub;m:"subscriptiondefault not defined for segment/segments ",string[x]," "]]}each errors;
      };
 
 initdatastripe:{
      .lg.o[`init;"init datastriping"];
      configload[];
      };
 
+
 \d .stpps
 
-// Function to map the where clause from config table extracted by .stpps.segmentfilter function to tablename
-// Allows use of ` as argument for tables
+//makes a dictionary of tables and there filters for segmentedsubdetails
+
 filtermap:{[tabs;id] if[tabs~`;tabs:.stpps.t]; ((),tabs)!.stpps.segmentfilter\:[(),tabs;id]}
 
-// Find where clause from config tables
+//grabs filters from the config files and for the "ignoretable" filter converts to "" to allow segmentedsudetails to run
+
 segmentfilter:{[tbl;segid]
-     wcref:first exec wcRef from .stpps.segmentconfig where table=tbl , segmentID=segid;
-     .stpps.segmentfiltermap[wcref]
+     id:`$string segid;
+     filter:first (flip .stpps.stripeconfig[id])[tbl];
+     $[filter~"ignoretable";filter:"";filter]
      };
 
-// Subscribe to particular segment using segmentID based on .u.sub
+//subscribe to a table using segmentID to determine filtering
 subsegment:{[tbl;segid];
-     //tablename and segmentid used to get filters
+//casting segid to an symbol as json is restrictive
+     id:`$string segid;
+     if[not (id in (key .stpps.stripeconfig));
+       .lg.e[`sub;m:"Segment ",string[segid]," is not defined in striping.json"];:()];
+     ignoredtables:`$();
+//setting the default for non-configured tables
+     default:.stpps.segmentfilter[`subscriptiondefault;segid];
      if[tbl~`;:.z.s[;segid] each .stpps.t];
-     if[not tbl in .stpps.t;
-          .lg.e[`sub;m:"Table ",string[tbl]," not in list of stp pub/sub tables"];
-          :();
-     ];
-     filter:segmentfilter[tbl;segid];
-     if[filter~"";
-          .lg.e[`sub;m:"Incorrect pairing of table ",string[tbl]," and segmentID ",string[segid]," not found in .stpps.segmentconfig"];
-          :();
-     ];
-     
+     stripedtables:inter [key flip .stpps.stripeconfig[id];.stpps.t];
```

Comment:
```suggestion
     stripedtables:.stpps.t inter key flip .stpps.stripeconfig[id];
```

personal preference, but IMO the infix notation is more readable in this case?

Reply:
Yes no problem do not how i keep missing them